### PR TITLE
IEP-486: GH #315: SDK Configuration editor do not recognize Kconfig "menuconfig ...." line and it parameters.

### PR DESCRIPTION
**Pull request by sigmaaa**
PR summary:
- rendering menuconfig elements as bool type, so the user is able to enable/disable them
- by enabling/disabling menuconfig elements, corresponding menus will be added/removed in the treeview. 
- highlighting menuconfig elements by adding the icon.

before:
menuconfig element is missing, no way to enable/disable it 
Screenshot 2021-11-09 at 10 49 28

after:
disabled menuconfig
Screenshot 2021-11-09 at 10 45 43
enabled menuconfig
Screenshot 2021-11-08 at 17 47 57

### CODE REVIEW

File: bundles/com.espressif.idf.sdk.config.ui/src/com/espressif/idf/sdk/config/ui/SDKConfigurationEditor.java [hunk 1]
```diff
@@ -611,7 +613,10 @@ protected void renderMenuItems(KConfigMenuItem selectedElement)
 					: false);
 			Object newConfigValue = modifiedJsonMap.get(configKey);
 			String helpInfo = kConfigMenuItem.getHelp();
-
+			if (type.equals(IJsonServerConfig.MENU_TYPE))
+			{
+				isVisible = true;
+			}
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.sdk.config.ui/src/com/espressif/idf/sdk/config/ui/SDKConfigurationEditor.java:619]: Can we move this validation here? https://github.com/espressif/idf-eclipse-plugin/blob/master/bundles/com.espressif.idf.sdk.config.core/src/com/espressif/idf/sdk/config/core/KConfigMenuItem.java#L107-L118

File: bundles/com.espressif.idf.sdk.config.ui/src/com/espressif/idf/sdk/config/ui/SDKConfigurationEditor.java [hunk 2]
```diff
@@ -648,6 +653,40 @@ else if (isVisible && type.equals(IJsonServerConfig.HEX_TYPE))
...
+					@Override
+					public void widgetSelected(SelectionEvent e)
+					{
+						JSONObject jsonObj = new JSONObject();
+						jsonObj.put(configKey, button.getSelection());
+						executeCommand(jsonObj);
+						renderMenuItems(kConfigMenuItem);
+					}
+
+				});
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.sdk.config.ui/src/com/espressif/idf/sdk/config/ui/SDKConfigurationEditor.java:677]: As this code is the same for `IJsonServerConfig.BOOL_TYPE`, could you extract to a method and utilize it.

**Review by kolipakakondal — CHANGES_REQUESTED** [review]

File: bundles/com.espressif.idf.sdk.config.core/src/com/espressif/idf/sdk/config/core/KConfigMenuProcessor.java [hunk 3]
```diff
@@ -84,7 +84,10 @@ protected void read(Object obj, KConfigMenuItem menuItem)
 			childMenu.setId((String) jsonObject.get("id")); //$NON-NLS-1$
 			childMenu.setType((String) jsonObject.get("type")); //$NON-NLS-1$
 			childMenu.setHelp((String) jsonObject.get("help")); //$NON-NLS-1$
-
+			if (jsonObject.get("is_menuconfig") != null) {
+				childMenu.setIsMenuConfig((boolean) jsonObject.get("is_menuconfig")); //$NON-NLS-1$
+			}
```
Comment (alirana01) [inline bundles/com.espressif.idf.sdk.config.core/src/com/espressif/idf/sdk/config/core/KConfigMenuProcessor.java:89]: We can create private static constant variable for these json keys. Or you can use the //$NON-NLS-1$ and //$NON-NLS-2$ if there are more than two hardcoded strings in one statement

**Review by alirana01 — COMMENTED** [review]
Just a minor change required if feasible else LGTM.

File: bundles/com.espressif.idf.sdk.config.ui/src/com/espressif/idf/sdk/config/ui/SDKConfigurationEditor.java (hunk 1 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.sdk.config.ui/src/com/espressif/idf/sdk/config/ui/SDKConfigurationEditor.java]: done

File: bundles/com.espressif.idf.sdk.config.core/src/com/espressif/idf/sdk/config/core/KConfigMenuProcessor.java (hunk 3 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.sdk.config.core/src/com/espressif/idf/sdk/config/core/KConfigMenuProcessor.java]: done

**Pull request merged**